runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6754,6 +6754,23 @@ TR_J9ByteCodeIlGenerator::genNewArray(int32_t typeIndex)
         }
      }
 
+   // Special case for handling Arrays.copyOfRangeByte when the caller of caller is String<init>(AbstractStringBuilder, Void).
+   // The caller is always Arrays.copyOfRange since Arrays.copyOfRangeByte is a private method.
+   if ((comp()->getInlineDepth() > 1) && !comp()->isPeekingMethod() && !generateArraylets
+       && _methodSymbol->getRecognizedMethod() == TR::java_util_Arrays_copyOfRangeByte)
+      {
+      int32_t callerIndex = comp()->getCurrentInlinedCallSite()->_byteCodeInfo.getCallerIndex();
+      int32_t callerOfCallerIndex = comp()->getInlinedCallSite(callerIndex)._byteCodeInfo.getCallerIndex();
+      TR::ResolvedMethodSymbol *callerOfCaller = callerOfCallerIndex > -1 ? comp()->getInlinedResolvedMethodSymbol(callerOfCallerIndex) : comp()->getOptimizer()->getMethodSymbol();
+
+      if(callerOfCaller->getRecognizedMethod() == TR::java_lang_String_init_AbstractStringBuilder_Void)
```

Comment:
I only check if the call chain is `String(ASB, Void) -> Anything -> copyOfRangeByte` the assumption is that the mission is to convert String builder to string when that pattern is observed but it may not be a safe assumption!

Reply:
Can you elaborate bit more, what you meant by this assumption ?

Reply:
I mean the purpose of String(ASB, Void) is to convert StringBuilder to String and the size of the resulting string can not be more than the string builder value. Therefore, when I see `copyOfRangeByte` with second caller `String(ASB, Void)` , I assume it is safe to skip zero init.